### PR TITLE
Add Repository Interface detection

### PR DESCRIPTION
**Pull request by hafriedlander**

### CODE REVIEW

File: code/model/DNProject.php [hunk 1]
```diff
@@ -621,6 +621,78 @@ public function getRepositoryURL() {
...
+	 */
+	public function getRepositoryInterfaceName() {
+		$url = parse_url($this->CVSPath);
+		$interfaces = $this->config()->repository_interfaces;
+		$host = strtolower($url['host']);
+
+		if (array_key_exists($host, $interfaces)) {
+			$interface = $interfaces[$host];
+
+			return isset($interface['name']) ? $interface['name'] : ucfirst(preg_split('/\./', $host)[0]);
```
Comment (micmania1) [inline code/model/DNProject.php]: `preg_split('/\./', $host)[0]` I don't think you can do this in 5.3

File: code/model/DNProject.php [hunk 2]
```diff
@@ -621,6 +621,78 @@ public function getRepositoryURL() {
...
+	 * {configuration} can either be boolean true to auto-detect both the host and the
+	 * name of the UI provider, or a nested array that overrides either one or both
+	 * of the auto-detected valyes
+	 *
+	 * @var array
+	 */
+	static private $repository_interfaces = array(
+		'github.com' => true,
+		'bitbucket.org' => true,
+		'repo.or.cz' => array(
```
Comment (micmania1) [inline code/model/DNProject.php]: is this supposed to be here?

File: code/model/DNProject.php (hunk 1 above)
Comment (hafriedlander) [inline code/model/DNProject.php]: Ah, true. Will fix.

File: code/model/DNProject.php (hunk 2 above)
Comment (hafriedlander) [inline code/model/DNProject.php]: repo.or.cz is the third most common free git hosting service. Happy to take it out, but ideally this'd support all the common public hosts by default.

Comment (micmania1) [inline code/model/DNProject.php]: fair enough. I've obviously been living under a rock :)

Comment (hafriedlander) [inline code/model/DNProject.php]: The opposite - it's _super_ old (long before github) :).

**Pull request merged**